src/components/form/DraggableForm.tsx

```diff
@@ -117,15 +124,46 @@ class DraggableForm extends PureComponent<Props, State> {
     }
 }
 
-const mapDispatchToProps = (dispatch, { id }: Props) => ({
-    addItem: (component: Component) => dispatch(addViewConfigNode(component)),
-    addChild: (childId) => dispatch(addViewConfigChild(id, childId)),
-    sortChild: (oldIndex, newIndex) => dispatch(sortViewConfigChild(id, oldIndex, newIndex))
+const mapDispatchToProps = (dispatch, { id, configType, ...rest }: Props) => {
+    if (configType === 'view') {
+        return {
+            sortItem: (oldIndex: number, newIndex: number) => dispatch(sortViewConfigChild(id, oldIndex, newIndex)),
+            addChild: (childId) => dispatch(addViewConfigChild(id, childId)),
+            addItem: (component: Component) => dispatch(addViewConfigNode(component))
+        };
+    } else {
+        return {
+            addItem: (facetConfig: FacetConfig) => dispatch(addFacetConfigItem(facetConfig)),
+            sortItem: (oldIndex: number, newIndex: number) => dispatch(sortFacetConfigItem(oldIndex, newIndex)),
+        };
+    }
+};
+
+const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps: OwnProps): Props => ({
+    ...stateProps,
+    ...dispatchProps,
+    ...ownProps,
+    addItem: (item: ConfigurableItem) => {
```

Comment:
Aren't you overwriting the addItem declared in mapDispatchToProps for the same check (configType === 'view')?

Reply:
Yea, this is by design.
dispatchProps's addItem only adds an item, this overload is there to also add it is a child immediately so the component does not need to do that itself.
I wanted to put as much of the logic as possible into the mapping functions so the component itself can be agnostic to what type of item it's actually editing.